ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,306 @@
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return True
+
+
+def _create_health_check(properties):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    for key, value in properties.iteritems():
+        health_check[key] = value
+
+    health_check = HealthCheck(**health_check)
+    return health_check
+
+
+def _create_load_balancer_params():
+    params_dict = {}
+    params_dict['listeners'] = ctx.node.properties['listeners']
+    params_dict['name'] = ctx.node.properties['elb_name']
+    params_dict['zones'] = ctx.node.properties['zones']
+    if 'security_groups' in ctx.node.properties.keys():
+        params_dict['security_groups'] = ctx.node.properties['security_groups']
+    return params_dict
+
+
+def _create_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to Create Load Balancer.')
+
+    params_dict = _create_load_balancer_params()
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.create_load_balancer(**params_dict)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer not created'
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('Load Balancer Created. ')
+
+    utils.set_external_resource_id(value=params_dict['name'],
+                                   ctx_instance=ctx.instance,
+                                   external=False)
+
+    ctx.instance.runtime_properties['elb_name'] = params_dict['name']
+
+    if not lb:
+        raise NonRecoverableError('Load Balancer not created')
+    # ctx.instance.runtime_properties['instance_list'] = []
+
+    return lb
+
+
+def _get_instance_list():
+
+    ctx.logger.debug('Attempting to get Load Balancer Instance List.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    try:
+        lb = _get_existing_load_balancer(elb_name)
+        list_of_instances = lb.instances
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer instance list not loaded'
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('retrieved Load Balancer  {0} Instance List. '
+                    ).format(elb_name)
+
+    ctx.instance.runtime_properties[constants.EXTERNAL_RESOURCE_ID] = lb.name
+
+    return list_of_instances
+
+
+def _get_load_balancers_by_names(list_of_names):
+
+    ctx.logger.debug(
+        'Attempting to get Load Balancers {0}.'.format(list_of_names))
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
```

Comment:
why not? it is using an external resource (boto config) and therefore can fail without it being our fault.